OSDOCS#6030: cert-manager: setting log levels (#59777)
openshift/openshift-docs: modules/cert-manager-enable-operator-log-level.adoc

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-log-levels.adoc

:_content-type: PROCEDURE
[id="cert-manager-enable-operator-log-level_{context}"]
= Setting a log level for the {cert-manager-operator}

You can set a log level for the {cert-manager-operator} to determine the verbosity of the operator log messages.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* The {cert-manager-operator} is installed.

.Procedure

. Update the subscription object for {cert-manager-operator} to provide the verbosity level for the operator logs by running the following command:
+
[source,terminal]
----

> $ oc -n cert-manager-operator patch subscription cert-manager-operator --type='merge' -p '{"spec":{"config":{"env":[{"name":"OPERATOR_LOG_LEVEL","value":"v"}]}}}' <1>

Reviewer: Good catch, @xingxingxia

----
<1> Replace `v` with the desired log level number. The valid values for `v` are `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, and `10`. The default value is `2`.

.Verification

> * Verify that the log level of the {cert-manager-operator} is updated by running the following command:
> +
> [source,terminal]
> ----
> $ oc logs deploy/cert-manager-operator-controller-manager -n cert-manager-operator

Reviewer 2: ```suggestion
* The pod of {cert-manager-operator} will be redeployed. Verify that the log level of the {cert-manager-operator} is updated by running the following commands:
+
[source,terminal]
----
$ oc set env deploy/cert-manager-operator-controller-manager -n cert-manager-operator --list | grep -e OPERATOR_LOG_LEVEL -e container
# deployments/cert-manager-operator-controller-manager, container kube-rbac-proxy
OPERATOR_LOG_LEVEL=9
# deployments/cert-manager-operator-controller-manager, container cert-manager-operator
OPERATOR_LOG_LEVEL=9

$ oc logs deploy/cert-manager-operator-controller-manager -n cert-manager-operator
```

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-log-levels.adoc

:_content-type: PROCEDURE
[id="cert-manager-enable-operator-log-level_{context}"]
= Setting a log level for the {cert-manager-operator}

> You can set a log level for the {cert-manager-operator} to determine the amount of detail in logged error messages. The valid values for the {cert-manager-operator} log level are `Normal`, `Debug`, `Trace`, and `TraceAll`.

Reviewer: Also would suggest, removing: `The valid values for the {cert-manager-operator} log level are Normal, Debug, Trace, and TraceAll.` as for operator log level we're just using numeric verbosity and not these enum values.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* The {cert-manager-operator} is installed.

.Procedure

> . Update the deployment for the {cert-manager-operator} to use the config map by running the following command:

Reviewer: Would prefer removing `config map` as we're just patching the subscription object and making no changes to any config map at this step.

Reviewer: How about something like: `Update the subscription object specific to {cert-manager-operator} in order to provide the verbosity level for the operator logs:`

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-log-levels.adoc

:_content-type: PROCEDURE
[id="cert-manager-enable-operator-log-level_{context}"]
= Setting a log level for the {cert-manager-operator}

You can set a log level for the {cert-manager-operator} to determine the verbosity of the operator log messages.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.

> * The {cert-manager-operator} 1.11.1 or later is installed.

Reviewer 3: Same comment as the previous module.

.Procedure

* Update the subscription object for {cert-manager-operator} to provide the verbosity level for the operator logs by running the following command:
+
[source,terminal]
----
$ oc -n cert-manager-operator patch subscription openshift-cert-manager-operator --type='merge' -p '{"spec":{"config":{"env":[{"name":"OPERATOR_LOG_LEVEL","value":"v"}]}}}' <1>
----

> <1> Replace `v` with the desired log level number. The valid values for `v` are `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, and `10`. The default value is `2`.

Reviewer 3: Just a suggestion
```suggestion
<1> Replace `v` with the desired log level number. The valid values for `v` can range from `1`to `10`. The default value is `2`.
```
